### PR TITLE
Wrap install.sh in a function

### PR DESCRIPTION
**Pull request by tianon**
With tabs, instead of spaces, and a comment about why we're wrapped up in a function. :wink:

Closes #12241

### CODE REVIEW

File: hack/install.sh [hunk 1]
```diff
@@ -20,213 +20,218 @@ command_exists() {
...
+		https://docs.docker.com/en/latest/installation/
 
-    https://docs.docker.com/en/latest/installation/
+	EOF
+	exit 1
+}
 
-EOF
-exit 1
+# wrapped up in a function so that we have some proection against only getting half the file during "curl | sh"
```
Comment (thaJeztah) [inline hack/install.sh]: Typo s/proection/protection

File: hack/install.sh [hunk 2]
```diff
@@ -20,213 +20,218 @@ command_exists() {
...
+			[ "$user" != 'root' ] && your_user="$user"
+			echo
+			echo 'If you would like to use Docker as a non-root user, you should now consider'
+			echo 'adding your user to the "docker" group with something like:'
+			echo
+			echo '  sudo usermod -aG docker' $your_user
+			echo
+			echo 'Remember that you will have to log out and back in for this to take effect!'
+			echo
+			exit 0
```
Comment (thaJeztah) [inline hack/install.sh]: (unrelated to this PR) Should this message also be wrapped in a function? The exact same message is included twice in this script (yeah, nit-picking)

Comment (tiborvass) [inline hack/install.sh]: +1

**Comment by tianon** [comment]
:+1: updated with a function for the shared message, and updated the larger uses of repeated `echo >&2` to use `cat >&2 <<-EOF` appropriately instead.

**Comment by thaJeztah** [comment]
Thanks!

**Comment by tianon** [comment]
Updated! :+1:

**Comment by tiborvass** [comment]
LGTM

**Comment by estesp** [comment]
LGTM

**Pull request merged**